File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) 
    {
        int r = 0;
        int l = height.size()-1;
        int max = 0;
        while(r < l)
        {
            int check = min(height[r],height[l]);
            if(check*abs(r-l) >= max)
            {
                max = check*abs(r-l);
            }
            if(height[r] > height[l])
            {
                l--;
            }
            else
            {
                r++;
            }
        }    
        return max;
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i = 0
        j = len(height) - 1

        ar = 0

        while(i < j):
            ar = max(ar,min(height[i],height[j]) * (j-i))
            if height[i] > height[j]:
                j -= 1
            else:
                i += 1
        
        return ar